components/public-api-server/pkg/oidc/router.go

```diff
@@ -169,13 +169,24 @@ func (s *Service) getCallbackHandler() http.HandlerFunc {
 				return
 			}
 		} else {
-			cookies, _, err := s.createSession(r.Context(), result, config)
+			cookies, message, err := s.createSession(r.Context(), result, config)
 			if err != nil {
 				log.WithError(err).Warn("Failed to create session from downstream session provider.")
 				reportLoginCompleted("failed", "sso")
 				respondeWithError(rw, r, "We were unable to create a user session.", http.StatusInternalServerError, useHttpErrors)
 				return
 			}
+			if message.NewUser {
```

Comment:
Ooph. Why do we have to loop the "newUser" bool through the OIDC flow, incl. injecting it into the returnUrl? Why not resolve it on the dashboard, e.g. based on localStorage and/or cookies?

Reply:
@geropl AFAIK the only place that really knows if we're dealing with a new or existing user is the iam session manager here: https://github.com/gitpod-io/gitpod/blob/f35abe827331d75bbc60950985206b11136ba8da/components/server/src/iam/iam-session-app.ts#L77-L79

I think I'm not following about the "Why not resolve it on the dashboard" part.